app/src/main/java/com/pose/move/network/ApiOperationWrapper.kt

```diff
@@ -0,0 +1,42 @@
+package com.pose.move.network
+
+import android.util.Log
+import com.squareup.moshi.Moshi
+import java.io.IOException
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.withContext
+import retrofit2.HttpException
+import kotlin.coroutines.CoroutineContext
+
+class ApiOperationWrapper @Inject constructor(moshi: Moshi) {
```

Comment:
I'd suggest moving this into an Interceptor. Same logic, but you won't have to explicitly call it every time you do a network operation.

The only part that can be moved is the `withContext` one, but even so, the whole ErrorWrapping mechanism could be easily moved into an interceptor.